pkg/daemon/controller/controller.go

```diff
@@ -134,28 +135,68 @@ func (sc *ServiceController) Bootstrap(desiredVersion string) error {
 }
 
 // RunController is the entry point of WICD's controller functionality
-func RunController(ctx context.Context, watchNamespace, kubeconfig, caBundle string) error {
-	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+func RunController(ctx context.Context, watchNamespace, kubeconfig, caBundle string, certDir, certDuration string) error {
+	var cfg *rest.Config
+	var err error
+
+	duration, err := time.ParseDuration(certDuration)
```

Comment:
If there is no reason for this function to have a string representation of time, please do this when parsing the cmd line flag and pass in a time.Duration to this function instead.